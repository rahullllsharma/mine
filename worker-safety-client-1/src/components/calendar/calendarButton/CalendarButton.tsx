import type { PropsWithClassName } from "@/types/Generic";
import cx from "classnames";

export type CalendarButtonProps = PropsWithClassName<{
  isActive?: boolean;
  isToday?: boolean;
  isDisabled?: boolean;
  onClick: () => void;
}>;

const disabledStyles = "disabled:opacity-30 disabled:cursor-not-allowed";

export default function CalendarButton({
  isActive,
  isToday,
  isDisabled,
  onClick,
  children,
  className,
}: CalendarButtonProps): JSX.Element {
  const isDefaultBorder = !isToday && !isActive;
  const borderStyles = cx("border rounded-xl border rounded-xl", {
    ["border-brand-gray-40"]: isDefaultBorder,
    ["border-4"]: isActive,
    ["border-brand-urbint-40"]: isToday,
    ["border-4 border-brand-gray-70"]: isActive && !isToday,
  });

  const attributes: { [key: string]: boolean } = {};
  if (isToday) {
    attributes["data-is-today"] = true;
  }
  if (isActive) {
    attributes["data-is-selected"] = true;
  }

  return (
    <button
      disabled={isDisabled}
      className={cx(
        "h-20 w-20 flex flex-col items-center justify-center",
        className,
        disabledStyles,
        borderStyles
      )}
      onClick={onClick}
      {...attributes}
    >
      {children}
    </button>
  );
}
