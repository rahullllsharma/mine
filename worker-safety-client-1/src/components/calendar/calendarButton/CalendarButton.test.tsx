import { render, screen, fireEvent } from "@testing-library/react";
import CalendarButton from "./CalendarButton";

const mockOnClick = jest.fn();

describe(CalendarButton.name, () => {
  it("should render a default calendar button", () => {
    const { asFragment } = render(<CalendarButton onClick={mockOnClick} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with a blue border when "isToday" is set to true', () => {
    const { asFragment } = render(
      <CalendarButton onClick={mockOnClick} isToday />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with a ticker and darker border when "isActive" is set to true', () => {
    const { asFragment } = render(
      <CalendarButton onClick={mockOnClick} isActive />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with a ticker and blue border when "isActive" and "isToday" are set to true', () => {
    const { asFragment } = render(
      <CalendarButton onClick={mockOnClick} isActive isToday />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should be disabled when "isDisabled" is set to true', () => {
    render(<CalendarButton onClick={mockOnClick} isDisabled />);
    const buttonElement = screen.getByRole("button");
    expect(buttonElement).toBeDisabled();
  });

  it('should call the "mockOnClick" callback when "onClick" event is triggered', () => {
    render(<CalendarButton onClick={mockOnClick} />);
    const buttonElement = screen.getByRole("button");
    fireEvent.click(buttonElement);
    expect(mockOnClick).toBeCalled();
  });
});
