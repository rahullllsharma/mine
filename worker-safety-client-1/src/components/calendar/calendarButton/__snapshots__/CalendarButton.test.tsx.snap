// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarButton should render a default calendar button 1`] = `
<DocumentFragment>
  <button
    class="h-20 w-20 flex flex-col items-center justify-center disabled:opacity-30 disabled:cursor-not-allowed border rounded-xl border rounded-xl border-brand-gray-40"
  />
</DocumentFragment>
`;

exports[`CalendarButton should render with a blue border when "isToday" is set to true 1`] = `
<DocumentFragment>
  <button
    class="h-20 w-20 flex flex-col items-center justify-center disabled:opacity-30 disabled:cursor-not-allowed border rounded-xl border rounded-xl border-brand-urbint-40"
    data-is-today="true"
  />
</DocumentFragment>
`;

exports[`CalendarButton should render with a ticker and blue border when "isActive" and "isToday" are set to true 1`] = `
<DocumentFragment>
  <button
    class="h-20 w-20 flex flex-col items-center justify-center disabled:opacity-30 disabled:cursor-not-allowed border rounded-xl border rounded-xl border-4 border-brand-urbint-40"
    data-is-selected="true"
    data-is-today="true"
  />
</DocumentFragment>
`;

exports[`CalendarButton should render with a ticker and darker border when "isActive" is set to true 1`] = `
<DocumentFragment>
  <button
    class="h-20 w-20 flex flex-col items-center justify-center disabled:opacity-30 disabled:cursor-not-allowed border rounded-xl border rounded-xl border-4 border-4 border-brand-gray-70"
    data-is-selected="true"
  />
</DocumentFragment>
`;
